Created workbench tutorial in mantid basics course. (#25754)
mantidproject/mantid: docs/source/tutorials/mantid_basic_course/using_workbench/03_Working_with_Python_in_Workbench.rst

.. _03_Working_with_Python_in_Workbench:

================================
Working with Python in Workbench
================================

Within the Mantid Workbench you can make use of the script window or IPython prompt to manipulate data and build plots exactly to your specifications in ways that you would not 
be able to with the interface alone.

.. figure:: /images/Workbench_script_new.png
   :width: 700px
   :alt: A view of the Workbench script window.
   
You can also use the IPython tab below the script editor to change to an inline Python interpreter. Both the python interpreter and scripts run thorugh Mantid are able to interact 
with Mantid workspaces, plot and algorithms.

1. Default imports
==================

By default any new script opened in Workbench will come with some default imports:

.. code-block:: python

   from __future__ import (absolute_import, division, print_function, unicode_literals)
   
This provideds compatability between scripts written in Python2 and Python3, notably 
all print statements must not use brackets.

.. code-block:: python

   from mantid.simpleapi import *
   

> This imports all the algorithems from Mantid to be used within your Python script.

Reviewer: ```suggestion
This imports all the algorithms from Mantid to be used within your Python script.
```

.. code-block:: python

   import matplotlib.pyplot as plt
   import numpy as np

> This imports Matplotlib for use as well the the excellent NumPy package.

Reviewer: ```suggestion
This imports Matplotlib for use as well as the excellent NumPy package.
```

2. Tabs and multiple scripts
============================

To open a new script you can click on the **+** button, or to open an existing script in a new tab go to “File” -> “Open Script” in the Workbench taskbar or use the hotkey `Ctrl + o`.

> From the script window you can also tab different scripts working on each independanty and multiple scripts can be run simultainiusly. Workbench will restore any tabs that were open

Reviewer: ```suggestion
From the script window you can also tab different scripts working on each independently and multiple scripts can be run simultaneously. Workbench will restore any tabs that were open
```

the last time Mantid was shut down.

3. Running, aborting, and options
===================================
From here you can run or abort your script, toggle options for accesability, and use the find and replace tool.

.. figure:: /images/Workbench_script_options.png
   :width: 700px
   :alt: A view of the Workbench script options.

* Run: This executes the currently selected text in your script, if nothing is selected it will run the entire script. The green play button serves as a shortcut for this option.
* Run All: This will run the entire script regardless of whether any text is selected.

> * Abort: This imediatly aborts the currently running script. The red stop button serves as a shortcut for this option.

Reviewer: ```suggestion
* Abort: This immediately aborts the currently running script. The red stop button serves as a shortcut for this option.
```
